amazonfreertossdk/src/main/java/software/amazon/freertos/amazonfreertossdk/AmazonFreeRTOSDevice.java

```diff
@@ -557,6 +555,7 @@ public void onCharacteristicRead(BluetoothGatt gatt,
                                 try {
                                     mMtu = Integer.parseInt(currentMtu.mtu);
                                     mMaxPayloadLen = mMtu - 3;
+                                    mMaxPayloadLen = Math.max(mMaxPayloadLen, 0);
```

Comment:
Above line could be removed by `mMaxPayloadLen = Math.max((mMtu - 3), 0 );`

Reply:
@ravibhagavandas where does the "3" come from here?

Reply:
Its the max payload length for BLE which is MTU - 3 bytes (for ATT header).